### ISSUE TITLE
Maintenence based APCs cannot be replaced.

### ISSUE THREAD

**Issue by Poojawa** [tgstation/tgstation#40169]
![](https://i.gyazo.com/63ce2c594a96f0043e5b6fab722c7e62.png)

Maintenance based APCs are not counting the wall area when being placed down. 

You switch the experimental welding tool on.
You start welding the APC frame...
You cut the APC frame from the wall.
*attempt to click wall with the frame cut*
This area already has an APC!

**Comment by 00aquir** [comment]
What about making it possible to connect the maint wire directly to the APC on the wall, without any terminals? Otherwise you must circumnavigate the areas every time because often you cannot place APC at the same place, many rooms have more than 1 APC on their wall.

It could be programmed so that you would be able to click on the APC with a wire while you have an exposed wire under yourself, and that would make it possible to connect it to the APC. Wouldn't that work?

**Comment by Poojawa** [comment]
....?

That's not the issue. The issue is you cannot replace APCs if destroyed from their mapped start in maintenance, Area Power Controller and all.

**Comment by 00aquir** [comment]
You can replace a room's APC which is situated in maint if you stand right inside the maintenance door's tile but the game won't let you do it for the same wall if you are standing in the maintenance! The terminal portion of it must belong to the same area the APC is at, otherwise you cannot replace it.

**Comment by Poojawa** [comment]
right. which is the bug that I'm reporting as an issue. by clicking the wall to place the APC, which is in the desired area, is where it should be taking its area info from.

the alternative fix is someone going about and extending the area position to encompass the terminal + APC. 

Sorry, it can be hard to follow what you're saying <3

**Comment by 00aquir** [comment]
I remember when, in previous versions, the radiation storm hit the terminal inside maintenance, because it belonged to the room's APC. That was a problem.

**Comment by Poojawa** [comment]
yup. but I would say that being able to repair an APC destroyed by a grief ghost role antag like Nightmares is better than losing a tile of Maint to a rad infusion.

**Comment by Poojawa** [comment]
and, of course, wrong button. hue

**Comment by 00aquir** [comment]
Well this is why I said that, if possible, the APC should directly connect to the wire, without the terminal.

**Comment by Barhandar** [comment]
They actually directly connect to the wire. The terminal is in the same tile as APC, APC is just visually shifted onto the wall.

What it actually should have, is step in its dir and take area from there, i.e. same thing it already does on map init.

**Comment by 00aquir** [comment]
well in that case yeah the APC should be on the actual wall

**Comment by Barhandar** [comment]
Please learn how BYOND and SS13 work before saying something this stupid. 
If it's on the wall tile it will be accessible from both sides of it.

**Comment by 00aquir** [comment]
I see what you mean but it can't be programmed so that it grants access only from directions it faces?

**Comment by Poojawa** [comment]
Relaaaax my dude @Barhandar . don't be mean to new folks.

@00aquir that's how it works in a nutshell, really. Hitboxes are what you click on. The offset makes it 'look' like it's attached to the wall. Due to how the game handles placement, it has to be on the same tile as you're placing it down on.

what I ended up finding out for the fix, was that the code was looking for the person placing the frame down and in what specific area they were in - which they're standing in maintenance! that's awfully silly. Maintenance obviously has its own APC! So I switch it to making it realize that what I wanted it to check was the wall itself that I placed it on, not the floor I was standing on - and it's fixed! 

https://github.com/tgstation/tgstation/pull/40186 check out that link, and look at the 'files changed' part to see how that silly little mislabel caused all of it.
